lib/common.js

```diff
@@ -320,7 +320,7 @@ function percentEncode(str) {
 }
 
 function matchStringOrRegexp(target, pattern) {
-  var str = !_.isUndefined(target) && target.toString ? target.toString() : target;
+  var str = !_.isUndefined(target) ? (target.toString ? target.toString() : target) : '';
```

Comment:
It's a bit difficult to understand this double conditional expression. Would it be clearer as an `if () ... else if () ... else`?